pluggy/hooks.py

```diff
@@ -257,20 +257,32 @@ def __call__(self, *args, **kwargs):
                 )
         return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
 
-    def call_historic(self, proc=None, kwargs=None):
-        """ call the hook with given ``kwargs`` for all registered plugins and
+    def call_historic(self, result_callback=None, kwargs=None, proc=None):
+        """Call the hook with given ``kwargs`` for all registered plugins and
         for all plugins which will be registered afterwards.
 
-        If ``proc`` is not None it will be called for for each non-None result
-        obtained from a hook implementation.
+        If ``result_callback`` is not ``None`` it will be called for for each
+        non-None result obtained from a hook implementation.
+
+        .. note::
+            The ``proc`` argument is now deprecated.
         """
-        self._call_history.append((kwargs or {}, proc))
+        if proc is not None:
+            warnings.warn(
+                "Support for `proc` argument now deprecated and will be"
+                "removed in an upcoming release.",
+                DeprecationWarning
+            )
+            result_callback = proc
+
+        self._call_history.append((kwargs or {}, result_callback))
         # historizing hooks don't return results
         res = self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
-        if proc is None:
+        if result_callback is None:
             return
+        # XXX: woah isn't this wrong for `firstresult` hooks?
```

Comment:
@nicoddemus btw did you notice this?
If i'm reading this correctly won't this code never process `firstresult` hooks?

Reply:
Yeah I think that loop expr is just slightly wrong should be `for x in res or [res]:` right?

Reply:
Oh waitttt [whatttt](https://github.com/pytest-dev/pluggy/blob/master/pluggy/hooks.py#L35)?
Wait why not?